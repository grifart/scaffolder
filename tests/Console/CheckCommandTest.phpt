<?php

declare(strict_types=1);

namespace Grifart\ClassScaffolder\Test\Console;

use Grifart\ClassScaffolder\Console\CheckCommand;
use Symfony\Component\Console\Command\Command;
use Symfony\Component\Console\Input\ArrayInput;
use Symfony\Component\Console\Output\BufferedOutput;
use Tester\Assert;

require __DIR__ . '/../bootstrap.php';

/**
 * @testCase
 */
final class CheckCommandTest extends CommandTestCase
{
	public function testCheck(): void
	{
		$input = new ArrayInput(['check', 'definition' => __DIR__ . '/check']);
		$output = new BufferedOutput();

		$exitCode = $this->runCommand($input, $output);

		Assert::same(1, $exitCode);
		Assert::matchFile(__DIR__ . '/CheckCommandTest.expected.txt', $output->fetch());
	}

	protected function createCommand(): Command
	{
		return new CheckCommand();
	}
}

(new CheckCommandTest())->run();
