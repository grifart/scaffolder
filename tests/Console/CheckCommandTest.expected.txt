Checking 1 definition file:

F


check/.definition.php: 3 definitions, 2 errors
----------------------------------------------

FAIL ModifiedClass

 [ERROR] RuntimeException: The generated file contains changes that will be lost
         if you generate it again.

         in %A%


FAIL MissingClass

 [ERROR] RuntimeException: There is no generated file for given definition.

         in %A%
